lib/Alcaeus/MongoDbAdapter/AbstractCursor.php

```diff
@@ -266,6 +280,12 @@ protected function ensureIterator()
     private function wrapTraversable(\Traversable $traversable)
     {
         foreach ($traversable as $key => $value) {
+            if ($this instanceof \MongoCursor &&
```

Comment:
Since this is logic that applies only to MongoCursor, we should probably not have it in AbstractCursor but instead move it to MongoCursor (which might require making this method protected and marking it as `@internal`).